spec/user_spec.rb

```diff
@@ -16,12 +16,113 @@
     expect(@subject['link_color']).to eq(@subject.user['link_color'])
   end
 
-  context '#cleanup_old_versions!' do
-    let(:user_keys) { { 'link_color:1' => 'blue' } }
+  describe '#cleanup_old_versions!' do
```

Comment:
I would perhaps rephrase the It blocks such that they are not conditional and should more directly describe what they are/are not doing. Here it appears the describes appear to more accurately reflect the assertions made by their inner Its.

Reply:
> We're modifying the implementation of 3 functions: `cleanup_old_experiments!`, `cleanup_old_versions` and `active_experiments`. Looking briefly at the specs there appears to be some existing issues with spec coverage, should we consider adding specs to handle the scenario where keys_without_finished_and_time_of_assignment has been replaced with the new function `experiment_keys`

added tests coverage for `cleanup_old_versions`, `active_experiments` and `max_experiments_reached?`. `cleanup_old_experiments` will be covered in [#86339](https://github.com/clio/themis/issues/86339).